pkg/net/neighborhood.go

```diff
@@ -149,50 +156,25 @@ func (n *neighborhood) newEmitter(p goprocess.Process, ch chan<- EvtState) emitt
 
 func (emit emitter) Emit(id peer.ID, ev Event) { emit(id, ev) }
 
-func (n *neighborhood) Handle(ctx context.Context, log Logger, h host.Host, s network.Stream) {
-	var ss sampleStep
-	if err := ss.RecvPayload(s); err != nil {
-		log.WithError(err).
-			Debug("error encountered during random walk (recv payload)")
-		return
-	}
-
-	// HACK:  the local peer can become orphaned during a walk
-	//		  if the join stream terminates concurrently.  In
-	//		  such cases, we pretend the remote peer is still a
-	//		  member of the neighborhood, and proceed as planned.
-	peer := s.Conn().RemotePeer()
-	if e, ok := n.vtx.Random(); ok {
-		peer = e.ID()
-	}
-
-	ctx, cancel := context.WithDeadline(ctx, ss.Deadline())
-	defer cancel()
-
-	if err := ss.Next(s, h, peer).Step(ctx); err != nil {
-		log.WithError(err).
-			Debug("error encountered during random walk (next step)")
-	}
-}
-
-func (n *neighborhood) Peers() []*peer.PeerRecord {
+func (n *neighborhood) Records() recordSlice {
 	es := n.vtx.Load()
-	peers := make([]*peer.PeerRecord, len(es))
+	records := make(recordSlice, len(es))
 
 	i := 0
-	for p := range es {
-		peers[i] = peer.NewPeerRecord()
-		peers[i].PeerID = p
-		// TODO: Addrs is filled outside this function, by the caller
-		i++
+	for _, e := range es {
+		records[i] = e.Record()
 	}
-	return peers
+	return records
 }
 
-func (n *neighborhood) MaxSize() int {
+func (n *neighborhood) MaxLen() int {
 	return 5 // TODO: decide maximum neighbor amounts
```

Comment:
You mean to refactor from a method to a const, so that `MaxLen` is not a method, or only the `5`? I made it a method just in case in the future the `MaxLen` becomes a dynamic value based on the environment

Reply:
I meant just the `5`.  The `MaxLen()` method seems like the right move, since we might tweak this at some point.

Reply:
Nice